web/modules/custom/dpl_admin/assets/dpl_admin.css

```diff
@@ -176,6 +176,20 @@
   }
 }
 
+/**
+  We use the password policy module, which shows it's own password restrictions.
+  We want to hide the standard drupal password suggestions, as they are
+  confusing, as they may give conflicting information.
+  We need to make sure that inline styling also gets overriden, hence [style]
+  and !important.
+ */
+.user-form {
+  .password-suggestions,
+  .password-suggestions[style] {
+    display: none !important;
+  }
+}
```

Comment:
Please reconsider this solution.

Hiding the elements via CSS seems somewhat crude to me. Have you considered doing it in the existing form alter?